pkg/apis/actuation/types.go

```diff
@@ -8,6 +8,8 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
is this needed for this PR? It can probably be pulled out.

Reply:
The refactor now requires `Inventory` to satisfy `runtime.Object`  which was previously not needed, hence why deepcopy-gen tags are added in this PR. We could spin it out into another PR that is merged first, but this tag addition is related to the refactor. LMKYT